### PR TITLE
Developer

### PR DESCRIPTION
**Pull request by alex-arabadjiev**
# Pull Request Template

## Description

Fixed a bug in which a coupon code entered when before logging in was lost after logging in.

## Type of change

- [X] Bug fix
- [ ] New feature
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I updated the CHANGELOG.md

### CODE REVIEW

File: src/Backend/SgateShopgatePlugin/Components/User.php [hunk 1]
```diff
@@ -80,9 +80,22 @@ public function loginUser($request, $httpResponse)
...
+             *
+             * All session variables are saved and replaced once the session
+             * is recreated with the desired id.
+             */
+            $oldSession = array_merge(array(), $_SESSION);
+            session_commit(); // Ends the current session
+            session_id($sessionId); // Sets the ID for the new session
+            session_start(); // Stars a fresh session with the new ID
+            $_SESSION = array_merge(array(), $oldSession); // Replace old session variables
+
```
Comment (alex-arabadjiev) [inline src/Backend/SgateShopgatePlugin/Components/User.php:97]: The session ID was being incorrectly set which caused the previous cart items to be lost.

File: src/Backend/SgateShopgatePlugin/Components/User.php [hunk 2]
```diff
@@ -123,15 +136,6 @@ public function loginUser($request, $httpResponse)
...
         } else {
-            if (!empty($basket['content'])) {
-                $this->basket->clearBasket();
-                $this->basket->sRefreshBasket();
-
-                foreach ($basket['content'] as $basketItem) {
-                    $this->basket->sAddArticle($basketItem['ordernumber'], $basketItem['quantity']);
-                }
-            }
-
```
Comment (alex-arabadjiev) [inline src/Backend/SgateShopgatePlugin/Components/User.php:134]: Once the cart-loss bug was fixed, this workaround for restoring its contents could be removed

File: src/Backend/SgateShopgatePlugin/Helpers/WebCheckout.php [hunk 3]
```diff
@@ -85,19 +85,6 @@ public function loginAppUser($token, $request)
...
-            if (!empty($basket['content'])) {
-                foreach ($basket['content'] as $basketItem) {
-                    Shopware()->Modules()->Basket()->sAddArticle($basketItem['ordernumber'], $basketItem['quantity']);
-                }
-
-                if (!empty($voucher)) {
-                    Shopware()->Modules()->Basket()->sAddVoucher($voucher['code']);
-                }
-            }
-
```
Comment (alex-arabadjiev) [inline src/Backend/SgateShopgatePlugin/Helpers/WebCheckout.php:100]: This cart-restoring workaround is also no longer necessary

**Review by Rekhyt — APPROVED** [review]

**Comment by Rekhyt** [comment]
Merging this so we can publish a new version of the plugin. I was told that there's more changes to come from, so feel free to open another PR as you see fit.

**Pull request merged**